src/traces/surface/defaults.js

```diff
@@ -50,13 +50,19 @@ module.exports = function supplyDefaults(traceIn, traceOut, defaultColor, layout
         }
     }
 
-    coerce('lighting.ambient');
-    coerce('lighting.diffuse');
-    coerce('lighting.specular');
-    coerce('lighting.roughness');
-    coerce('lighting.fresnel');
-    coerce('hidesurface');
-    coerce('opacity');
+    //Coerce remaining properties
+    [
+        'lighting.ambient',
+        'lighting.diffuse',
+        'lighting.specular',
+        'lighting.roughness',
+        'lighting.fresnel',
+        'lightposition.x',
+        'lightposition.y',
+        'lightposition.z',
+        'hidesurface',
```

Comment:
Adding the `vertexnormalsepsilon` and `facenormalsepsilon` functionality to mesh3d traces only is fine. :+1: 

But in this case, we should **not** add `vertexnormalsepsilon` and `facenormalsepsilon` to the  `surface` attributes to not confuse users reading attribute reference [schema](https://plot.ly/javascript/reference/).

To do so:

- remove `vertexnormalsepsilon` and `facenormalsepsilon` from the `surface` [attributes](https://github.com/plotly/plotly.js/pull/556/files#diff-b4784b514eb9a0e0ada46bac7f483a64R210)
- rewrite this [line](https://github.com/plotly/plotly.js/pull/556/files#diff-a9cf6c3c9e5f641b36f80ac4f8cb2dc7R128) as

```js
lighting: extendFlat({}, surfaceAtts.lighting, {
  vertexnormalepsilon: {
    /* */
  }, 
  facenormalepsilon: {
    /* */
  }
})
```